CRM/Core/Invoke.php

```diff
@@ -367,6 +369,14 @@ public static function statusCheck($template) {
     $status = Civi::cache('checks')->get('systemStatusCheckResult');
     $template->assign('footer_status_severity', $status);
     $template->assign('footer_status_message', CRM_Utils_Check::toStatusLabel($status));
+
+    // show user a warning if CiviCRM is in explicit maintenance mode
+    // NOTE: if maintenance mode is inherited from the CMS, we expect the CMS
+    // to issue its own warning
+    $coreMaintenanceMode = \Civi::settings()->get('core_maintenance_mode');
+    if ($coreMaintenanceMode && ($coreMaintenanceMode !== 'inherit')) {
```

Comment:
Aah, clever way to determine if somebody else is providing warnings... :heavy_check_mark: